File: 118. Pascal Triangle/py/PascalTriangle.py
# LeetCode problem: 118. Pascal's Triangle
# https://leetcode.com/problems/pascals-triangle/description/


class Solution:
    def generate(self, numRows: int) -> list[list[int]]:
        # LeetCode constraint: 1 <= numRows <= 30
        tree = [[1]]
        for i in range(1, numRows):
            row = [1] + [tree[i-1][j-1] + tree[i-1][j] for j in range(1, i)] + [1]
            tree.append(row)
        return tree

File: 118. Pascal Triangle/cs/PascalTriangle/PascalTriangle.cs
// LeetCode problem: 118. Pascal's Triangle
// https://leetcode.com/problems/pascals-triangle/description/

namespace LeetCode.PascalTriangle;

public class Solution
{
    public IList<IList<int>> Generate(int numRows)
    {
        // LeetCode constraint: 1 <= NumRows <= 30
        IList<IList<int>> tree = [[1]];
        for (int i = 1; i < numRows; i++)
        {
            var row = new List<int> { 1 };
            for (int j = 1; j < i; j++)
                row.Add(tree[i - 1][j - 1] + tree[i - 1][j]);
            row.Add(1);
            tree.Add(row);
        }
        return tree;
    }
}